packages/delisp-core/src/convert.ts

```diff
@@ -293,6 +306,45 @@ defineToplevel("export", expr => {
   };
 });
 
+defineToplevel("type", expr => {
+  const [typeOp, ...args] = expr.elements;
+
+  if (args.length !== 2) {
+    const lastExpr = last([typeOp, ...args]) as ASExpr;
+    throw new Error(
+      printHighlightedExpr(
+        `'type' needs exactly 2 arguments, got ${args.length}`,
+        lastExpr.location,
+        true
+      )
+    );
+  }
+
+  const [name, definition] = args;
+
+  if (name.type !== "symbol") {
+    throw new Error(
+      printHighlightedExpr("'type' expected a symbol as a name", name.location)
+    );
+  }
+
+  checkUserDefinedTypeName(name);
+
+  const definitionType = convertType(definition);
+  if (listTypeVariables(definitionType).length > 0) {
+    throw new Error(
+      printHighlightedExpr("Type variable out of scope", definition.location)
```

Comment:
Why is this limitation required?

Reply:
I'm not sure honestly. I just checked that it is the case in Haskell. So I wanted to be cautious.

The meaning of it can be tricky. Actually, types aliases can contain variables. In a future PR, we could implement

`(type (Person a) {:name string | a})`

but for then we need to work on the kind system as well.